### ISSUE TITLE
[Bug] [Unhandled exception] UnicodeDecodeError

### ISSUE THREAD

**Issue by GitMensch** [OpenCobolIDE/OpenCobolIDE#204]
## Steps to reproduce

I've just opened a file which containes german umlauts, stored as plain ANSI...

## Traceback

```
  File "d:\documents\pyqode.core\pyqode\core\api\client.py", line 278, in _on_process_stderr_ready

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfc in position 948: invalid start byte
```

## System information

- OpenCobolIDE: 4.6.5
- GnuCOBOL: GNU Cobol 1.1.0
- Python: 3.4.3 (32bits)
- Qt: 5.5.0
- PyQt: 5.5
- pyqode.core: 2.6.8
- pyqode.cobol: 2.6.3
- pyqode.qt: 2.7.dev0
- pygments: 2.0.2
- QDarkStyle: 2.1

**Comment by GitMensch** [comment]
Notes: the log and main window shows encoding cp1252.

**Comment by ColinDuquesnoy** [comment]
I am glad I added this except hook ;-)

Here the client process is trying to decode the stderr from the backend server process which should be encoded using the system's default encoding.

I am probably not using the right method to retrieve this encoding (and I am using that wrong method everywhere I need to deal with system encoding, e.g. when decoding the compiler output). So this might indeed be related to #188 (not sure about #195).

What is the output of the following python script (just start the python3 command prompt and paste the below code):

```python
import locale
import sys
print(sys.getdefaultencoding())
print(locale.getpreferredencoding())
```

Also it would help if you could send me the file you used (or upload it somewhere).

**Comment by GitMensch** [comment]
You can create the file yourself (create new file from within OCIDE), here is the content:

~~~~~cobol
      *-----------------------------------------------------------------
      * 5555555555555555555555555555555555555555555555555
      * 6666666666666666666666666666666666666666666666666
      *-----------------------------------------------------------------
      *-----------------------------------------------------------------
       WORKING-STORAGE SECTION.
      *-----------------------------------------------------------------
       PROCEDURE DIVISION.
       HAUPT SECTION.
      *
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      *-
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***555555555555555555555555555555555555555555555555599
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      *
      ***1561sadfsadfsadfsadfsadfsadf safÜR sdgdsafgdfs
                 IF rrrrrrrrrrrrrrrr-BBBBWEB
                    OR aaaaaaaaaaaaaaaaaaa
      *-
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
                 END-IF
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
      ***9999999999999999999999999999999999999999999999
      ***5555555555555555555555555555555555555555555555555
              END-IF
           END-IF
      *
      *
           continue.
      *
      *sdag sdfh sjhfgj hsdh dsfh sdgfh   adsfsdf sd ätte
      *
           EXIT PROGRAM.
~~~~~

**Comment by ColinDuquesnoy** [comment]
I can finally reproduce this bug :) This happens only with the frozen application (the one I distribute on windows with python frozen into the app executable), the problem does not occur when I run the OCIDE using python (e.g. from the terminal or from hackedit). 

Probably a cx_freeze related issue (again).

> I am probably not using the right method to retrieve this encoding

Note for myself: I double checked that and no, I am using the right function: locale.getpreferredencoding() return cp1252 on my french system which is the expected result.

**Comment by ColinDuquesnoy** [comment]
Ok now this make sense. The bug was fixed in pyQode/pyqode.core@9839a6ffca2844b09fbd110ac75d8c10170a6413 on Aug 04. I actually fixed an issue I had at work with one of our internal tool that make use of pyqode and completely forget about it. I've just cherrypicked it into master to include it in the next bug fix release.

I was really using a hardcoded "utf-8" encoding to decode stderr bytes instead of ``locale.getpreferredencoding()``.

This issue has nothing to do with #195 and #188. I am using ``locale.getpreferredencoding()`` for decoding compiler output since v4.6.3 (commit f6f998f5315daf563b901d38058738d71d90c976).
